# Phân tích và xử lí dữ liệu về bệnh đái tháo đường
Tổng quan về tập dữ liệu:
1. Number of times pregnant : số lần có thai.
2. Plasma glucose concentration a 2 hours in an oral glucose tolerance test : bài kiểm tra độ tích tụ lượng đường trong cơ thể.
3. Diastolic blood pressure (mm Hg): chỉ số đo huyết áp tâm trương (áp lực động mạch khi tim nghỉ ngơi giữa 2 lần co bóp).
4. Triceps skin fold thickness (mm): độ dày da gắp nép của bắp tay sau.
5. 2-Hour serum insulin (mu U/ml): chỉ số đo insulin trong cơ thể sau 2 giờ dung nạp gluco
6. Body mass index (weight in kg/(height in m)^2): chỉ số đo tỉ lệ cân nặng chiều cao theo chuẩn BMI.
7. Diabetes pedigree function: chỉ số đo bệnh về đái tháo đường theo phả hệ.
8. Age (years): tuổi.

# Bắt đầu

## 1. Phân tích tập dữ liệu

### 1.1 thêm các thư viện cần thiết

In [3]:
import numpy as np
import pandas as pd
import matplotlib as plt
import IPython.display as display

### 1.2 Đọc tập dữ liệu 

In [4]:
data_path  = "data/pima-indians-diabetes.csv"
data_names = ["Pregnant","2Hr_Gluco_Concentrate","Blood_Pressure","Skin_Fold","2Hr_Isnulin","BMI","Predigy","Age","Class"]
df_dataset = pd.read_csv(data_path, names = data_names)

In [5]:
df_dataset.head()

Pregnant  2Hr_Gluco_Concentrate  Blood_Pressure  Skin_Fold  2Hr_Isnulin  \
0         6                    148              72         35            0   
1         1                     85              66         29            0   
2         8                    183              64          0            0   
3         1                     89              66         23           94   
4         0                    137              40         35          168   

    BMI  Predigy  Age  Class  
0  33.6    0.627   50      1  
1  26.6    0.351   31      0  
2  23.3    0.672   32      1  
3  28.1    0.167   21      0  
4  43.1    2.288   33      1

### 1.3 Kiểm tra tính toàn vẹn của tập dữ liệu

In [6]:
df_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Pregnant               768 non-null    int64  
 1   2Hr_Gluco_Concentrate  768 non-null    int64  
 2   Blood_Pressure         768 non-null    int64  
 3   Skin_Fold              768 non-null    int64  
 4   2Hr_Isnulin            768 non-null    int64  
 5   BMI                    768 non-null    float64
 6   Predigy                768 non-null    float64
 7   Age                    768 non-null    int64  
 8   Class                  768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


- Tập dữ liệu gồm 768 bản ghi không null và hợp lệ kiểu dữ liệu

In [7]:
df_dataset.isna().sum()

Pregnant                 0
2Hr_Gluco_Concentrate    0
Blood_Pressure           0
Skin_Fold                0
2Hr_Isnulin              0
BMI                      0
Predigy                  0
Age                      0
Class                    0
dtype: int64

In [10]:
(df_dataset == 0).sum()

Pregnant                 111
2Hr_Gluco_Concentrate      5
Blood_Pressure            35
Skin_Fold                227
2Hr_Isnulin              374
BMI                       11
Predigy                    0
Age                        0
Class                    500
dtype: int64

### 1.4 Thống kê trên dữ liệu số
- Count, Mean, Standard Deviation, Minimum Value
- 25th Percentile, 50th Percentile (Median), 75th Percentile, Maximum Value

In [10]:
display.display(df_dataset.drop(columns=["Class"]).describe().T)

count        mean         std     min       25%  \
Pregnant               768.0    3.845052    3.369578   0.000   1.00000   
2Hr_Gluco_Concentrate  768.0  120.894531   31.972618   0.000  99.00000   
Blood_Pressure         768.0   69.105469   19.355807   0.000  62.00000   
Skin_Fold              768.0   20.536458   15.952218   0.000   0.00000   
2Hr_Isnulin            768.0   79.799479  115.244002   0.000   0.00000   
BMI                    768.0   31.992578    7.884160   0.000  27.30000   
Predigy                768.0    0.471876    0.331329   0.078   0.24375   
Age                    768.0   33.240885   11.760232  21.000  24.00000   

                            50%        75%     max  
Pregnant                 3.0000    6.00000   17.00  
2Hr_Gluco_Concentrate  117.0000  140.25000  199.00  
Blood_Pressure          72.0000   80.00000  122.00  
Skin_Fold               23.0000   32.00000   99.00  
2Hr_Isnulin             30.5000  127.25000  846.00  
BMI                     32.0000   36.60000   67.10  
Predigy                  0.3725    0.62625    2.42  
Age                     29.0000   41.00000   81.00

#### Nhận xét: 